services/fileuploader/gzwriter.go

```diff
@@ -0,0 +1,52 @@
+package fileuploader
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+type MultiFileWriter interface {
+	Write(workspaceID, path string, data []byte) (int, error)
```

Comment:
This is a multi file writer, why bothering with workspaceIDs at all? We could rename it to `key` instead and keep it generic. Thoughts?

Reply:
Removed dependency on the workspaceID